src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -7968,9 +7968,145 @@ public static String rotate(final String str, final int shift) {
+         * Some VMs reserve some header words in an array.
+         * Attempts to allocate larger arrays may result in
+         * OutOfMemoryError: Requested array size exceeds VM limit
+         */
+        private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8;
+
+        private String[] elementData = DEFAULTCAPACITY_EMPTY_ELEMENTDATA;
+        private int size = 0;
+
+        private static int hugeCapacity(int minCapacity) {
+            if (minCapacity < 0) {// overflow
+                throw new OutOfMemoryError();
+            }
+            return (minCapacity > MAX_ARRAY_SIZE)
+                    ? Integer.MAX_VALUE
+                    : MAX_ARRAY_SIZE;
+        }
+
+        String get(int index) {
+            return elementData[index];
+        }
+
+        void add(String e) {
+            add(e, elementData, size);
+        }
+
+        /**
+         * This helper method split out from add(E) to keep method
+         * bytecode size under 35 (the -XX:MaxInlineSize default value),
+         * which helps when add(E) is called in a C1-compiled loop.
+         */
+        private void add(String e, Object[] elementData, int s) {
+            if (s == elementData.length) {
+                elementData = grow(size + 1);
+            }
+            elementData[s] = e;
+            size = s + 1;
+        }
+
+        /**
+         * Increases the capacity to ensure that it can hold at least the
+         * number of elements specified by the minimum capacity argument.
+         *
+         * @param minCapacity the desired minimum capacity
+         * @throws OutOfMemoryError if minCapacity is less than zero
+         */
+        private String[] grow(int minCapacity) {
+            String[] newElementData = new String[newCapacity(minCapacity)];
+            System.arraycopy(elementData, 0, newElementData, 0, elementData.length);
+            return elementData = newElementData;
+        }
+
+        /**
+         * Returns a capacity at least as large as the given minimum capacity.
+         * Returns the current capacity increased by 50% if that suffices.
+         * Will not return a capacity greater than MAX_ARRAY_SIZE unless
+         * the given minimum capacity is greater than MAX_ARRAY_SIZE.
+         *
+         * @param minCapacity the desired minimum capacity
+         * @throws OutOfMemoryError if minCapacity is less than zero
+         */
+        private int newCapacity(int minCapacity) {
+            // overflow-conscious code
+            int oldCapacity = elementData.length;
+            int newCapacity = oldCapacity + (oldCapacity >> 1);
+            if (newCapacity - minCapacity <= 0) {
+                if (elementData == DEFAULTCAPACITY_EMPTY_ELEMENTDATA) {
+                    return Math.max(DEFAULT_CAPACITY, minCapacity);
+                }
+                if (minCapacity < 0) { // overflow
+                    throw new OutOfMemoryError();
```

Comment:
We shouldn't use this error type, this is expected to be thrown by the JVM

>Thrown when the Java Virtual Machine cannot allocate an object because it is out of memory

Reply:
I will edit this later.